File: interpreter-pattern/python/src/Context.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

# 构建可执行环境上下文
class Context:
  def __init__(self, key, value):
    self.map = dict()
    self.add(key, value)
  
  def __init__(self):
    self.map = dict()

  def add(self, key, value):
    self.map[key] = value
  

  def get(self, key):
    return self.map[key]
  

File: interpreter-pattern/go/src/Context.go
package src

// 构建可执行环境上下文
type Context struct {
  Map map[string]int
}

func (c *Context) Init(key string, value int) {
  c.Map = make(map[string]int)
  c.Add(key, value)
}

func (c *Context) Add(key string, value int) {
  if c.Map == nil {
    c.Map = make(map[string]int)
  }
  c.Map[key] = value
}

func (c *Context) Get(key string) int {
  return c.Map[key]
}
